Network exercise integration

jleiv01 opened the pull request:
-Added 4 Exercises with Networking in mind, added check sums to the list.
-Updated README on main. 
-Updated terraform main file on both azure and aws directories, this was to add a network rule for port 80.
-Tested both on Azure and AWS (had an issue with setup.sh on AWS, had a misspelling which cause some delay, its now fixed)

#### Challenge 9: DNS troubleshooting

DNS troubleshooting is an important skill, fix the issue and find the flag.

#### Challenge 10: Remote upload

Use a remote transfer application to transfer a file to the `ctf_challenges` directory and receive your flag 

#### Challenge 11: Web Configuration

Find and fix the issue, access the site from your local browser by using the provided IP from Terraform. 

#### Challenge 12: Network Traffic Analysis

Use network capture tools to review generated network traffic and find its flag in the sent message.


Hope there are no syntax or errors on this PR.

madebygps commented:
Just did a QA on azure side:

9 works 
10 does not accept the flag, does not verify (maybe hash issue)?
11 works
12 works

<img width="427" alt="Screenshot 2025-02-17 at 4 14 06 PM" src="https://github.com/user-attachments/assets/6d59c1d4-4d8f-46bc-bd43-a4df68514824" />

@rishabkumar7 please QA aws version

jleiv01 commented:
Corrected the issue, wrong hash was on the initial commit on question 10, thanks for the QA. Also updated the ctf_setup.sh file on the AWS side, it contained the same issue.

rishabkumar7 commented:
QA for AWS:
- Challenge 9: fix dns resolution ✅ 
- Challenge 10: Remote upload ❌  probably something with the flag hashing still
<img width="398" alt="Screenshot 2025-02-18 at 12 48 07 PM" src="https://github.com/user-attachments/assets/b1e4813d-b500-42c4-9837-d134f3331048" />

- Challenge 11: Web Configuration ✅ 
<img width="547" alt="Screenshot 2025-02-18 at 10 42 34 AM" src="https://github.com/user-attachments/assets/dee41ff8-394e-46d1-9676-41472f8d98ba" />

- Challenge 12: Network Traffic Analysis ✅ 

<img width="305" alt="Screenshot 2025-02-18 at 12 39 41 PM" src="https://github.com/user-attachments/assets/38efa37a-9b11-412b-99c4-bb0fc794e59b" />

@jleiv01 could you look into the Challenge 10.

madebygps commented:
@rishabkumar7  did you qa on latest commit? according to discord he pushed an update to resolve issue

jleiv01 commented:
@madebygps @rishabkumar7 thanks, let me know if the latest commit

`88d1c56f4568965c61817492aa8580e5e3ff982c`


Shows up for you with the fix for #10 on this PR. 

I had the wrong hash on the 1st commit.

rishabkumar7 commented:
> @rishabkumar7 did you qa on latest commit? according to discord he pushed an update to resolve issue

@madebygps yeah, I think so, but will double check and make sure.

madebygps commented:
i just qa the most recent commit, everything works fine on  the azure side, only change for flag 12 please make it all lowercase @jleiv01 

@rishabkumar7 please qa aws side

rishabkumar7 reviewed (changes requested):
@jleiv01 could you make the flag for 12 challenge lowercase, like others.
everything else works.

jleiv01 commented:
@madebygps @rishabkumar7 I have updated both AWS and Azure setup files, changed Flag number 12 to lower case and updated its hash.

rishabkumar7 commented:
thanks @jleiv01 just tested and it works, the flag is lowercase.

great work on this PR. 🚀
